packages/core/src/interpreter.ts

```diff
@@ -135,8 +137,9 @@ export class Interpreter<
     const { clock, logger, parent, id } = resolvedOptions;
     const self = this;
 
-    // TODO: this should come from a "system"
-    this.sessionId = registry.bookId();
+    this.system = parent?.system ?? options?.system ?? createSystem();
```

Comment:
why do we allow `system` to be provided through options? Shouldn't it always be given by the parent or instantiated here?

Reply:
Yes it should - it was originally provided through options for this use-case:

```ts
const system = createSystem(...);

const actor = interpret(machine, { system }).start();
```

But I don't think that `createSystem(...)` is in scope for this PR anymore.